pkg/configuration/base/resources/proxy.go

```diff
@@ -0,0 +1,124 @@
+package resources
+
+import (
+	"context"
+
+	apiconfigv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	openshiftGlobalProxy = "cluster"
+	envHTTPProxyName     = "HTTP_PROXY"
+	envHTTPSProxyName    = "HTTPS_PROXY"
+	envNoProxyName       = "NO_PROXY"
+)
+
+var (
+	ProxyAPIAvailable   = false
+	ProxyAPIChecked     = false
+	allProxyEnvVarNames = []string{
+		envHTTPProxyName,
+		envHTTPSProxyName,
+		envNoProxyName,
+	}
+	ProxyEnvVars = []corev1.EnvVar{}
+)
+
+func IsOpenshiftProxyAPIAvailable(clientSet *kubernetes.Clientset) bool {
+	if ProxyAPIChecked {
+		return ProxyAPIAvailable
+	}
+	gv := schema.GroupVersion{
+		Group:   apiconfigv1.GroupName,
+		Version: apiconfigv1.SchemeGroupVersion.Version,
+	}
+	if err := discovery.ServerSupportsVersion(clientSet, gv); err != nil {
+		// error, API not available
+		ProxyAPIChecked = true
+		ProxyAPIAvailable = false
+	} else {
+		// API Exists
+		ProxyAPIChecked = true
+		ProxyAPIAvailable = true
+	}
+	return ProxyAPIAvailable
+}
+
+func QueryProxyConfig(manager manager.Manager) error {
+	global := &apiconfigv1.Proxy{}
+	err := manager.GetAPIReader().Get(context.TODO(), client.ObjectKey{Name: openshiftGlobalProxy}, global)
+	if err != nil {
+		logger.Info("openshift global proxy not found")
+		return err
+	}
+
+	// We have found the global proxy configuration object!
+	ProxyEnvVars = ToEnvVar(global)
+	return nil
+}
+
+func ToEnvVar(proxy *apiconfigv1.Proxy) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  envHTTPProxyName,
```

Comment:
As this is a constant, can you rename it to uppercase only.
```
const (
 HTTP_PROXY = "HTTP_PROXY" 
)
```
would be ok.

Reply:
we should Camelcase and HTTPPROXY is ugly, lets keep it